src/sources/journald.rs

```diff
@@ -552,31 +591,65 @@ impl Drop for RunningJournalctl {
     }
 }
 
-fn create_event(record: Record, batch: &Option<BatchNotifier>) -> LogEvent {
+fn create_event(
+    record: Record,
+    batch: &Option<BatchNotifier>,
+    log_namespace: LogNamespace,
+) -> LogEvent {
     let mut log = LogEvent::from_iter(record).with_batch_notifier_option(batch);
 
     // Convert some journald-specific field names into Vector standard ones.
     if let Some(message) = log.remove(MESSAGE) {
         log.insert(log_schema().message_key(), message);
     }
```

Comment:
This code seems kinda sketchy. Like shouldn't we always have a `message` field ?

Reply:
It looks like the initial record is deserialized from Json.  I presume that Json record has a field called `MESSAGE`. This is essentially renaming that field to be whatever `message_key()` is - generally `message`.

Reply:
Is the goal to change this so that it moves the message to the root object similar to how the `ByteDeserializer` works - and thus make it conform to the schema spec from the `ByteDeserializerConfig` that is called in `schema_definition` here?

Reply:
I _think_ that feels right, these events are still being created in a JSON style though - AFAICT.

Bytes decoding would break the legacy path, if I'm understanding correctly.

Reply:
Yeah I should have been more clear in my comment- I understand this code is renaming the decoded journalctl message key into our `message` key.
What I thought was odd about it was that we have an if conditional, but don't do anything if the log didn't contain `MESSAGE` ... which I would think is an error? I guess if it's not an error case that bit of code could have ideally had a comment explaining that.

> Is the goal to change this so that it moves the message to the root object similar to how the ByteDeserializer works - and thus make it conform to the schema spec from the ByteDeserializerConfig that is called in schema_definition here?

I had no goal other than to point out that I intentionally didn't modify these lines of code since I did not see us doing that in other log namespacing PRs yet. But wanted to draw attention to the lack of error handling I mentioned above, and also if there is something we should be doing with it regarding the log namespacing.

I _think_ I follow your line of thought with moving it to the root object. It seems like that would maybe only be something we'd want to do in the Vector namespace though ?